docs/src/tutorials/yocto-linux.md

```diff
@@ -51,61 +50,191 @@ process. If using these options, to see previous commits or other branches see
 [here](https://stackoverflow.com/questions/29270058/how-to-fetch-all-git-history-after-i-clone-the-repo-with-depth-1)
 and [here](https://stackoverflow.com/questions/17714159/how-do-i-undo-a-single-branch-clone).
 
+> Alternatively, you could use `--branch=honister` for Yocto version 3.4 Honister. If doing so, remember to also use
+> `--branch=honister` for all additional layers that require it.
+
+```bash
+git clone git://git.yoctoproject.org/poky --branch=kirkstone --depth=1
+```
+
+Resulting directory structure should look like this:
+
+```
+.
+└── poky
+    ├── bitbake
+    ├── contrib
+    ├── documentation
+    ├── LICENSE
+    ├── LICENSE.GPL-2.0-only
+    ├── LICENSE.MIT
+    ├── MAINTAINERS.md
+    ├── Makefile
+    ├── MEMORIAM
+    ├── meta
+    ├── meta-poky
+    ├── meta-selftest
+    ├── meta-skeleton
+    ├── meta-yocto-bsp
+    ├── oe-init-build-env
+    ├── README.hardware.md -> meta-yocto-bsp/README.hardware.md
+    ├── README.md -> README.poky.md
+    ├── README.OE-Core.md
+    ├── README.poky.md -> meta-poky/README.poky.md
+    ├── README.qemu.md
+    └── scripts
+```
+
+### Initialize the build environment
+
+To use the `bitbake`, `bitbake-layers`, `runqemu`, or other tools used in yocto to create or run our build, we need to
+source the `poky/oe-init-build-env` script in our shell:
+
 ```
-$ git clone git://git.yoctoproject.org/poky --branch=kirkstone --depth=1
+cd poky
+source oe-init-build-env
 ```
 
+The script creates a `build` directory, which itself contains only a `conf` directory:
+
+```
+.
+└── poky
+    ├── build
+    │   └── conf
+    │       ├── bblayers.conf
+    │       ├── local.conf
+    │       └── templateconf.cfg
+    ...
+```
+
+Inside the `poky/build/conf` directory, there are 2 files of interest to us:
+
+- `bblayers.conf` contains all the layers used by build
+- `local.conf` contains local build configuration, ie. configuration that applies only to the build
+
+In the following steps, we will edit these files to customize our image.
+
 ### Add layers
 
-We'll be using `meta-tedge` and `meta-openembedded`. First, fetch the repositories:
+Clone the `meta-tedge` and `meta-openembedded` repositories to our working directory:
```

Comment:
In the above step, the script will take into `build` directory. It's not clear where to clone these two repos.

Reply:
That's right, I'll clarify that the script `cd`'s to `build` directory and to `cd` back to the main directory before performing this step.